pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -505,15 +506,14 @@ func convertResolution(proxyType model.NodeType, service *model.Service) cluster
 	case model.DNSRoundRobinLB:
 		return cluster.Cluster_LOGICAL_DNS
 	case model.Passthrough:
+		if service.Attributes.ServiceRegistry == provider.Kubernetes && features.EnableEDSForHeadless {
+			return cluster.Cluster_EDS
+		}
 		// Gateways cannot use passthrough clusters. So fallback to EDS
-		if proxyType == model.SidecarProxy {
```

Comment:
why changing this semantic?

Reply:
It is for easy readability. Personal preference but I can revert

Reply:
Nope, you changed, not refactor

Reply:
Can you check now - basically want to return early if it is gateway

Reply:
right now